core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushPredicateThroughProjectIntoWindow.java

```diff
@@ -199,12 +198,12 @@ private static OptionalInt extractUpperBound(TupleDomain<Symbol> tupleDomain, Sy
 
         Range span = values.getRanges().getSpan();
 
-        if (span.getHigh().isUpperUnbounded()) {
+        if (span.isHighUnbounded()) {
             return OptionalInt.empty();
         }
 
-        long upperBound = (Long) span.getHigh().getValue();
-        if (span.getHigh().getBound() == BELOW) {
+        long upperBound = (Long) span.getHighBoundedValue();
+        if (!span.isHighInclusive()) {
```

Comment:
should there be another method, `isHighExclusive`?

Reply:
maybe... idk